src_py/sprite.py

```diff
@@ -1099,17 +1099,14 @@ def draw(self, surface, bgd=None):
                     if spr._visible:
                         # sprite not dirty; blit only the intersecting part
                         _spr_rect = spr.rect
-                        if spr.source_rect is not None:
```

Comment:
I think a `DirtySprite`'s `rect` and `source_rect` can be different sizes so this code is probably still required to ensure the proper image is drawn.

Reply:
My understanding is that `source_rect` is used to get the correct image. If you look above at lines 1063-1067 you will see that a similar thing is being done there.